전남대 BE_조홍식_3주차 과제(2단계)

Daolove0323 opened the pull request:
기존 Long type id를 UUID로 변경하였고, 지난 step1에서 말씀하신 피드백 반영했습니다.
globalMapper 클래스의 메서드 네이밍은 step3에서 반영하도록 하겠습니다.
변경파일이 많아 보시기에 불편하실 듯해서, 과제중 궁금한 사항은 dm으로 보냈습니다!
항상 감사드립니다

Daolove0323 committed:
feat: step0 and code refactor

Daolove0323 committed:
fefac: 피드백반영

Daolove0323 committed:
feat: WishRequestDto 클래스 구현

Daolove0323 committed:
feat: memberRequestDto 구현

Daolove0323 committed:
feat: productResonse/request dto 구현

Daolove0323 committed:
refac: member 클래스 리팩토링

Daolove0323 committed:
refac: product 엔티티 리팩토링

Daolove0323 committed:
refac: wish 엔티티 리팩토링

Daolove0323 committed:
refac: memberRepository 리팩토링

Daolove0323 committed:
refac: productRepository 리팩토링

Daolove0323 committed:
refac: wishRepository 리팩토링

Daolove0323 committed:
refac: memberService 리팩토링

Daolove0323 committed:
refac: productService 리팩토링

Daolove0323 committed:
refac: wishService 리팩토링

Daolove0323 committed:
refac: wishService 리팩토링

Daolove0323 committed:
refac: memberController 리팩토링

Daolove0323 committed:
refac: 코드리팩토링

클래스/메서드명 변경에 따른 코드리팩토링

Daolove0323 committed:
Merge remote-tracking branch 'upstream/daolove0323' into step1

Daolove0323 committed:
refac: 피드백반영

코드 정리 피드백 반영,
옵셔널 관련 피드백 반영,
불필요한 PathVariable 파라미터 관련 피드백 반영,
Long type id를 UUID type으로 변경

Daolove0323 committed:
fix: Member : Wish (n:m) 연관관계 표현

Daolove0323 committed:
fix: Product : Wish (n:m) 연관관계 표현

Daolove0323 committed:
fix: Wish 엔티티 필드 수정

Daolove0323 committed:
feat: WishResponse Dto에서 사용할 memberDto 작성

Daolove0323 committed:
refac: wish 관련 dto 추가

Daolove0323 committed:
fix: wishService 연관관계 추가에 따른 리팩토링

Daolove0323 committed:
fix: WishController 연관관계 추가에 따른 리팩토링

Daolove0323 committed:
Squashed commit of the following:

commit cbe2cc1ed6858c1eca61d1ced20729a391bb7fe7
Author: Daolove0323 <90020593+Daolove0323@users.noreply.github.com>
Date:   Fri Jul 12 01:23:12 2024 +0900

    전남대 BE_조홍식 3주차 과제 (1단계) (#163)

    * feat: step0 and code refactor

    * fefac: 피드백반영

    * feat: WishRequestDto 클래스 구현

    * feat: memberRequestDto 구현

    * feat: productResonse/request dto 구현

    * refac: member 클래스 리팩토링

    * refac: product 엔티티 리팩토링

    * refac: wish 엔티티 리팩토링

    * refac: memberRepository 리팩토링

    * refac: productRepository 리팩토링

    * refac: wishRepository 리팩토링

    * refac: memberService 리팩토링

    * refac: productService 리팩토링

    * refac: wishService 리팩토링

    * refac: wishService 리팩토링

    * refac: memberController 리팩토링

    * refac: 코드리팩토링

    클래스/메서드명 변경에 따른 코드리팩토링

Daolove0323 committed:
Squashed commit of the following:

commit cbe2cc1ed6858c1eca61d1ced20729a391bb7fe7
Author: Daolove0323 <90020593+Daolove0323@users.noreply.github.com>
Date:   Fri Jul 12 01:23:12 2024 +0900

    전남대 BE_조홍식 3주차 과제 (1단계) (#163)

    * feat: step0 and code refactor

    * fefac: 피드백반영

    * feat: WishRequestDto 클래스 구현

    * feat: memberRequestDto 구현

    * feat: productResonse/request dto 구현

    * refac: member 클래스 리팩토링

    * refac: product 엔티티 리팩토링

    * refac: wish 엔티티 리팩토링

    * refac: memberRepository 리팩토링

    * refac: productRepository 리팩토링

    * refac: wishRepository 리팩토링

    * refac: memberService 리팩토링

    * refac: productService 리팩토링

    * refac: wishService 리팩토링

    * refac: wishService 리팩토링

    * refac: memberController 리팩토링

    * refac: 코드리팩토링

    클래스/메서드명 변경에 따른 코드리팩토링
Merge remote-tracking branch 'upstream/daolove0323' into step2

seungy0 commented on src/main/java/gift/controller/GlobalMapper.java:
```diff
@@ -0,0 +1,48 @@
+import gift.domain.Wish;
+import java.util.UUID;
+
+public class GlobalMapper {
```
이렇게 매퍼를 모으신 이유가 궁금합니다~
장단점이 있을 것 같습니다.

Daolove0323 replied:
service클래스에서 구현하기엔 역할에 맞지 않다고 생각이 들었습니다. 또 Entity와 Dto는 게터/세터 외의 메서드가 있으면 안된다고 생각해서 mapper 클래스를 구현했습니다. 각 Dto 클래스마다 mapper클래스를 만들면 너무 많아지는 느낌이 들어서 GlobalMapper 클래스를 구현하게 되었습니다!

seungy0 commented on src/main/java/gift/controller/auth/AuthController.java:
```diff
@@ -12,17 +12,31 @@
         this.authService = authService;
     }
 
+    public static void validateUserOrAdmin(LoginResponse member, UUID id) {
```
UUID를 사용하신 이유가 궁금합니다.

Daolove0323 replied:
UUID가 아닌 Long 타입을 사용하면서 AUTO INCREMENT를 하게 되면, 사용자 입장에서 너무 쉽게 예측할 수 있어 보안성이 떨어진다는 느낌을 받았습니다.

Daolove0323 committed:
fix: 불필요한 파일삭제 및 코드 정리

Daolove0323 committed:
fix: 버그 수정

Daolove0323 committed:
feat: 테스트 코드 작성

시나리오 테스트와 유닛테스트를 작성하였습니다.
실패하는 경우에 대한 검증도 하고 싶었지만, 반환형이 void은 validateAdmin() 메서드를 통해 토큰을 검증하다보니, globalHandler가 UnathorizedException을 적절히 핸들링해도 ResponseEntity에 담기 어려웠습니다. 이 방법에 대해서는 좀 더 고민하고 추가적인 테스트 코드를 작성하겠습니다..

seungy0 commented on src/main/java/gift/service/ProductService.java:
```diff
@@ -1,51 +1,54 @@
+    public ProductResponse update(UUID id, ProductRequest product) {
+        Product foundProduct = productRepository.findById(id)
+            .orElseThrow(ProductNotExistsException::new);
         foundProduct.setName(product.name());
         foundProduct.setPrice(product.price());
         foundProduct.setImageUrl(product.imageUrl());
```
```suggestion
        foundProduct.updateDetails(product.name(), product.price(), product.imageUrl());
```
이런식으로 교체하는 걸 추천드립니다.

Daolove0323 replied:
다음 과제에서 반영하겠습니다!

seungy0 commented on src/main/java/gift/domain/Member.java:
```diff
@@ -1,54 +1,64 @@
+    private String grade;
+
+    @OneToMany(mappedBy = "member",  cascade = CascadeType.REMOVE)
+    private final List<Wish> wishes = new LinkedList<>();
```
사용되지 않고 있는데, 필요한 부분일까요?

Daolove0323 replied:
일단 유지보수하면서 확장성을 고려해 양방향으로 구현했습니다

seungy0 commented on src/main/java/gift/domain/Product.java:
```diff
@@ -1,28 +1,31 @@
 public class Product {
+
+    @OneToMany(mappedBy = "product", cascade = CascadeType.REMOVE)
+    private final List<Wish> wishes = new LinkedList<>();
```
사용되지 않고 있는데, 필요한 부분일까요?

seungy0 reviewed (approved):
홍식님 고생하셨습니다.
몇가지 코멘트 남겼으니 참고해주세요.
